src/Http/Client.php

```diff
@@ -176,7 +176,7 @@ private function parseResponse(ResponseInterface $response)
             return null;
         }
 
-        if (!\in_array('application/json', $response->getHeader('content-type'), true)) {
+        if (!$this->isJSONResponse($response->getHeader('content-type'))) {
```

Comment:
Do we really need to check if it's json response? If we expect always to get json, then just delegate to json decoding, catch syntax error and just rethrow with meaningful message if needed?

Reply:
Yes, it should be always a JSON response, but I have no idea how we could do the way you're suggesting 😅

Reply:
Something like:

```diff
Index: src/Http/Client.php
===================================================================
diff --git a/src/Http/Client.php b/src/Http/Client.php
--- a/src/Http/Client.php	(revision a85bd4c4c54afc16f67c7bce66ccb789e810363e)
+++ b/src/Http/Client.php	(date 1721816875796)
@@ -176,16 +176,16 @@
             return null;
         }
 
-        if (!\in_array('application/json', $response->getHeader('content-type'), true)) {
-            throw new InvalidResponseBodyException($response, $response->getBody()->getContents());
-        }
-
         if ($response->getStatusCode() >= 300) {
             $body = $this->json->unserialize($response->getBody()->getContents()) ?? $response->getReasonPhrase();
 
             throw new ApiException($response, $body);
         }
 
-        return $this->json->unserialize($response->getBody()->getContents());
+        try {
+            return $this->json->unserialize($response->getBody()->getContents());
+        } catch (JsonDecodingException $e) {
+            throw new InvalidResponseBodyException($response, $response->getBody()->getContents());
+        }
     }
 }

```

Reply:
And I had a test that does not work as intended anymore:

```php
public function testInvalidResponseContentTypeThrowsException(): void
    {
        $httpClient = $this->createHttpClientMock(200, '<b>not json</b>', 'text/html');

        $client = new Client('https://localhost', null, $httpClient);

        $this->expectException(InvalidResponseBodyException::class);
        $this->expectExceptionMessage('Decoding payload to json');

        $client->get('/');
    }
```

In my mind the expectation before the exercise would catch that exception `InvalidResponseBodyException` because any other exception thrown before was swallowed by the `catch (Exception $e)`. So I didn't understand if this was only because the test was wrong or we would be changing the behavior if proceed with this change:

this is the error: 

`1) Tests\Http\ClientTest::testInvalidResponseContentTypeThrowsException
Failed asserting that exception of type "Meilisearch\Exceptions\JsonDecodingException" matches expected exception "Meilisearch\Exceptions\InvalidResponseBodyException". Message was: "Decoding payload to json failed: "Syntax error"." at`

Reply:
Maybe the exception has been raised before... Unfortunately, I don't have time to keep looking at this issue :|

Reply:
We could merge like this, then take a look at this later with more time, what do you think?